### ISSUE TITLE
OIDC - maybe allow id token for authentication

### ISSUE THREAD

**Issue by tomas-langer** [helidon-io/helidon#3590]
Similar to what was proposed in https://github.com/oracle/helidon/pull/3457

This requires some analysis whether such usage is the right way to go.

**Comment by jmntn2000** [comment]
See pull request #3637 

Tried to build off of the updates from @tomas-langer and add the following functionality to OIDC:

1) Added server side refresh.  If you are using cookies and server side OIDC provider you run into issues when the access token expires.  All javascript related calls from the client will fail due to expired token and js calls for SPA will not redirect to lean on OIDC Provider for new tokens.  To solve this I saved the refresh token encrypted in a separate cookie and when access token used for the request is expired I refresh it and update cookies so there is no disruption to the client (assuming refresh token is still valid).

2) Change the build subject to utilize the id token for principal if available.
